src/galaxy.cpp

```diff
@@ -722,6 +733,179 @@ void SelectPlayerView::clickPlayer(int x, int y, int arg) {
+			sprintf(buf, "%s %s", _game->_starSystems[planet.star].name, romanNumbers[planet.orbit]);
+
+			// Planet climate
+			switch(planet.climate) {
+			case TOXIC:
+				sprintf(buf, "%s", "Toxic");
+				asset_id = 26;
+				break;
+			case RADIATED:
+				sprintf(buf, "%s", "Radiated");
+				asset_id = 31;
+				break;
+			case BARREN:
+				sprintf(buf, "%s", "Barren");
+				asset_id = 36;
+				break;
+			case DESERT:
+				sprintf(buf, "%s", "Desert");
+				asset_id = 41;
+				break;
+			case TUNDRA:
+				sprintf(buf, "%s", "Tundra");
+				asset_id = 46;
+				break;
+			case OCEAN:
+				sprintf(buf, "%s", "Ocean");
+				asset_id = 51;
+				break;
+			case SWAMP:
+				sprintf(buf, "%s", "Swamp");
+				asset_id = 56;
+				break;
+			case ARID:
+				sprintf(buf, "%s", "Arid");
+				asset_id = 61;
+				break;
+			case TERRAN:
+				sprintf(buf, "%s", "Terran");
+				asset_id = 66;
+				break;
+			case GAIA:
+				sprintf(buf, "%s", "Gaia");
+				asset_id = 71;
+				break;
+			}
+			fnt->centerText(138, rowHeight * displayedPlanet - climateOffset, color, buf);
+			smallFnt->centerText(138, rowHeight * displayedPlanet + offset, color, "0 Food"); // FIXME: Pull real data
+
+			// Planet Gravity
+			switch(planet.gravity) {
+			case LOW_G:
+				sprintf(buf, "%s", "Low G");
+				break;
+			case NORMAL_G:
+				sprintf(buf, "%s", "Normal G");
+				break;
+			case HEAVY_G:
+				sprintf(buf, "%s", "Heavy G");
+				break;
+			}
+			fnt->centerText(220, rowHeight * displayedPlanet, color, buf);
+
+			// Minerals
+			switch(planet.minerals) {
+			case ULTRA_POOR:
+				sprintf(buf, "%s", "Ultra Poor");
+				break;
+			case POOR:
+				sprintf(buf, "%s", "Poor");
+				break;
+			case ABUNDANT:
+				sprintf(buf, "%s", "Abundant");
+				break;
+			case RICH:
+				sprintf(buf, "%s", "Rich");
+				break;
+			case ULTRA_RICH:
+				sprintf(buf, "%s", "Ultra Rich");
+				break;
+			}
+			fnt->centerText(306, rowHeight * displayedPlanet - mineralsOffset, color, buf);
+			smallFnt->centerText(306, rowHeight * displayedPlanet + offset, color, "3 prod/worker"); // FIXME: Pull real data
+
+			// Planet size
+			x = 50;
+			switch(planet.size) {
+			case TINY_PLANET:
+				sprintf(buf, "%s", "Tiny");
+				break;
+			case SMALL_PLANET:
+				sprintf(buf, "%s", "Small");
+				asset_id += 1;
+				x -= 1;
+				break;
+			case MEDIUM_PLANET:
+				sprintf(buf, "%s", "Medium");
+				asset_id += 2;
+				x -= 1;
+				break;
+			case LARGE_PLANET:
+				sprintf(buf, "%s", "Large");
+				asset_id += 3;
+				x -= 1;
+				break;
+			case HUGE_PLANET:
+				sprintf(buf, "%s", "Huge");
+				asset_id += 4;
+				x -= 4;
+				break;
+			}
+			fnt->centerText(385, rowHeight * displayedPlanet - sizeOffset, color, buf);
+			smallFnt->centerText(385, rowHeight * displayedPlanet + offset, color, "4 max pop"); // FIXME: Pull real data
+
+			// Planet icon
+			_planet = gameAssets->getImage(COLONY_VIEW_PLANET_LIST_ARCHIVE, asset_id, _planetAssets->palette());
```

Comment:
Load the assets in class constructor. Reloading them during every screen redraw is a waste of time.